<?xml version="1.0" encoding="utf-8"?>

<launch>

    <!--Launch USB Camera node-->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="1280" />
        <param name="image_height" value="720" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
    </node>

    <!--Launch Darknet YoloV3-->
    <arg name="launch_prefix" default=""/>
    <arg name="image" default="/usb_cam/image_raw" />
    <!-- Config and weights folder. -->
    <arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
    <arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>
    <!-- ROS and network parameter files -->
    <arg name="ros_param_file"             default="$(find darknet_ros)/config/ros.yaml"/>
    <arg name="network_param_file"         default="$(find darknet_ros)/config/yolov3-tiny-sud.yaml"/>
    <!-- Load parameters -->
    <rosparam command="load" ns="darknet_ros" file="$(arg ros_param_file)"/>
    <rosparam command="load" ns="darknet_ros" file="$(arg network_param_file)"/>
    <!-- Start darknet and ros wrapper -->
    <node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
        <param name="weights_path"          value="$(arg yolo_weights_path)" />
        <param name="config_path"           value="$(arg yolo_config_path)" />
        <remap from="/image_raw"  to="$(arg image)" />
    </node>

    <!--Launch SUD vision object-->
    <node pkg="sud2_moveit_config" name="sud_vision" type="sud_pouch_vision.py" output="screen"> 
    </node>
    
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/sud_detected_pouches"/>
        <param name="autosize" value="true" />
    </node>

    <!--Launch MT Scale Object-->
    <node pkg="sud2_moveit_config" name="MTScale" type="MTSerialScale.py" output="screen">
    </node>

</launch>